src/coreclr/vm/jitinterface.cpp

```diff
@@ -2240,6 +2240,25 @@ unsigned CEEInfo::getClassGClayout (CORINFO_CLASS_HANDLE clsHnd, BYTE* gcPtrs)
     return result;
 }
 
+int32_t CEEInfo::getModuleDataBlobOffset (int32_t* pInitFlagMask)
```

Comment:
I think this should take the class handle, and it should return the address of the init-flag for the given class handle, not offset.

Reply:
@jkotas hm.. at this point I only have a moduleId and classIndex (I guess in theory I can get a classhandle for them somehow)

Reply:
Can you keep track of the class handle in the helper node?

It may help if you do this together with changing the JIT helper that we call on the slow path to INITCLASS (https://github.com/dotnet/runtime/pull/47901#discussion_r571039913). We can import all ctor triggers as INITCLASS helpers that naturally track the class handle.

Reply:
@jkotas Ah, good point! Totally forgot about your suggestion with the single-argument INITCLASS function

Reply:
Addressed.